### ISSUE TITLE
"Restoring Previous State" bug; optimizations reset or do not save/work

### ISSUE THREAD

**Issue by TheRealestLad** [rcmaehl/NotCPUCores#40]
When I was trying to set the priority and core allocation for Steam.exe and looking at the debug console's commands it mentions "Exited. Restoring Previous State" after setting Steam.exe from turning it off/on and it resets back to the default settings. If I try to optimized Steam.exe while it is running the custom settings applied will work unless I turn off Steam.exe

Evidence for the 'exited' thing: https://imgur.com/a/GE3Xkp7
After restarting/deactivating Steam: https://imgur.com/a/MzNB5XX
While Steam is running: https://imgur.com/a/utApd3j

**Comment by rcmaehl** [comment]
So you're setting custom allocation for steam and it's getting reset to all cores after you restart steam or can you provide steps on how to reproduce?

**Comment by TheRealestLad** [comment]
> 
> 
> So you're setting custom allocation for steam and it's getting reset to all cores after you restart steam or can you provide steps on how to reproduce?

I am not that great at explaining on how to reproduce but let me try my best. What I do/did was:

1: Open up NotCPUCores
2: Put in Steam.exe into the process bar
3: Set the priority and core allocation
4: Click apply (open the debug console to see if you will get the 'restoring' message

alternate method:

1: Open up Steam (the app) and have it running
2: Open up NotCPUCores
3: Find the Steam.exe process in the debug console
4: One-click the steam process, and click the arrow on the right of the process bar/box
5: Set custom prioity and allocation
6: Close/restart Steam (Either NotCPUCores running in the background or not)
7: Open Steam again

But somehow if I try to set the affinity/priority of steam.exe via the notcpucores while steam is running and you go to the task manager and find the steam.exe process in the 'Details' and right click on steam.exe and click on the affinity or hover over priority the custom settings trying to be applied are shown and work, but if i restart steam.exe the custom settings will reset back to default state ('normal' priority and all cores being used)

**Comment by rcmaehl** [comment]
NotCPUCores doesn't automatically reapply custom priority/allocation once a program is relaunched, but I can definitely add that option in the next release. Next release should allow optimizing multiple processes at once as well.

**Comment by TheRealestLad** [comment]
> 
> 
> NotCPUCores doesn't automatically reapply custom priority/allocation once a program is relaunched, but I can definitely add that option in the next release. Next release should allow optimizing multiple processes at once as well.

Will you also try to make this work with when you restart/shutoff computer?

**Comment by rcmaehl** [comment]
> Will you also try to make the automatic re-applying work after when you restart/shutoff computer, then launching the prioritized, allocated app?

Yep, Auto-start, multiple process optimization, settings saving, and more are planned. Need to finish up the new UI first and iron out a lot of bugs that came with making the new UI.

**Comment by ThumpieBunnyEve** [comment]
+1 for the "resume prior saved process optimizations" at relaunch.
